.github/workflows/pretest.yml

```diff
@@ -58,3 +58,48 @@ jobs:
         ./generate.py --matrix matrix.yaml --schema schema.yaml --dry-run
         mypy generate.py
         popd
+
+  rocm-build:
+    runs-on: ubuntu-18.04
+    env:
+      CUPY_CI: "GitHub"
+      ROCM_REPO: "4.3.1"
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+
+    - name: Setup ROCm
+      run: |
+        wget -qO - https://repo.radeon.com/rocm/rocm.gpg.key | sudo apt-key add -
+        echo "deb [arch=amd64] http://repo.radeon.com/rocm/apt/${ROCM_REPO}/ ubuntu main" | sudo tee /etc/apt/sources.list.d/rocm.list
+        sudo apt-get -qqy update
+        sudo apt-get -qqy install rocm-dev hipblas hipfft hipsparse rocsparse rocrand rocthrust rocsolver rocfft hipcub rocprim rccl
+
+        echo "ROCM_HOME=/opt/rocm" >> "${GITHUB_ENV}"
+        echo "PATH=${ROCM_HOME}/bin:${PATH}" >> "${GITHUB_ENV}"
+        echo "LD_LIBRARY_PATH=${ROCM_HOME}/lib:${LD_LIBRARY_PATH}" >> "${GITHUB_ENV}"
+
+        hipconfig
```

Comment:
How about using an AMD provided rocm docker image?

Reply:
Raised another PR! Let's compare which one is faster 😄  https://github.com/cupy/cupy/pull/6031

Reply:
It seems this PR is slightly faster than #6034 

This PR: https://github.com/cupy/cupy/runs/4130093885
#6034: https://github.com/cupy/cupy/runs/4131011246